### ISSUE TITLE
SIGHUP is not ignored even with nohup

### ISSUE THREAD

**Issue by vicasong** [oracle/graal#5512]
We use `native-image` to package a tool to work in the background, but we found that even with `nohup`, the program exits when the session terminates, for this I made a simple replay scenario:

```bash
[cc@dev-base test]$ nohup ./tool sleep 30d &
[1] 22816
[cc@dev-base test]$ nohup: ignoring input and appending output to ‘nohup.out’

[cc@dev-base test]$ jobs -l
[1]+ 22816 Running                 nohup ./tool sleep 30d &
[cc@dev-base test]$ kill -SIGHUP 22816
[cc@dev-base test]$ jobs
[1]+  Exit 129                nohup ./tool sleep 30d
[cc@dev-base test]$
[cc@dev-base test]$ nohup sleep 300000000 &
[1] 23479
[cc@dev-base test]$ nohup: ignoring input and appending output to ‘nohup.out’

[cc@dev-base test]$ jobs -l
[1]+ 23479 Running                 nohup sleep 300000000 &
[cc@dev-base test]$ kill -SIGHUP 23479
[cc@dev-base test]$ jobs -l
[1]+ 23479 Running                 nohup sleep 300000000 &
[cc@dev-base test]$ kill 23479
[cc@dev-base test]$ jobs -l
[1]+ 23479 Terminated              nohup sleep 300000000
```
As above, I packaged a `tool` program and let it run for a long time in the background, but after sending a `SIGHUP` signal to it, the program exited. However, using the `sleep` command of linux is normal and can ignore `SIGHUP`.

This `tool` program is very simple, there is a sleep command:
```java
@CommandLine.Parameters(index = "0",
			paramLabel = "<Time>",
			description = "Sleep time.")
protected String time;

@Override
public Integer call() throws Exception {
		Duration duration;
		if (StringUtils.hasText(time)) {
			try {
				duration = DurationStyle.detectAndParse(time);
			}
			catch (IllegalArgumentException e) {
				printError(e.getMessage());
				return 1;
			}
		}
		else {
			duration = Duration.ofSeconds(30);
		}
		printInfo("Program going to sleep: %dms".formatted(duration.toMillis()));
		Thread.sleep(duration.toMillis());
		return 0;
}
```

At the same time, I registered `ShutdownHook` to output the call stack when exiting, and found this:
```text
Thread[SIGHUP handler,5,main]
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.headers.Pthread.pthread_cond_wait(Pthread.java)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixParkEvent.condWait(PosixPlatformThreads.java:323)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.parkCurrentPlatformOrCarrierThread(PlatformThreads.java:857)
        at java.base@17.0.5/jdk.internal.misc.Unsafe.park(Unsafe.java:48)
        at java.base@17.0.5/java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.monitor.JavaMonitorQueuedSynchronizer$ConditionNode.block(JavaMonitorQueuedSynchronizer.java:129)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.monitor.JavaMonitorQueuedSynchronizer$JavaMonitorConditionObject.await(JavaMonitorQueuedSynchronizer.java:473)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.monitor.MultiThreadedMonitorSupport.doWait(MultiThreadedMonitorSupport.java:331)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.monitor.MonitorSupport.wait(MonitorSupport.java:94)
        at java.base@17.0.5/java.lang.Object.waitSubst(Object.java:108)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.join(PlatformThreads.java:325)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.JavaThreads.join(JavaThreads.java:186)
        at java.base@17.0.5/java.lang.Thread.join(Thread.java:548)
        at java.base@17.0.5/java.lang.Thread.join(Thread.java:1372)
        at java.base@17.0.5/java.lang.ApplicationShutdownHooks.runHooks(ApplicationShutdownHooks.java:107)
        at java.base@17.0.5/java.lang.ApplicationShutdownHooks$1.run(ApplicationShutdownHooks.java:46)
        at java.base@17.0.5/java.lang.Shutdown.runHooks(Shutdown.java:130)
        at java.base@17.0.5/java.lang.Shutdown.exit(Shutdown.java:173)
        at java.base@17.0.5/java.lang.Terminator$1.handle(Terminator.java:51)
        at java.base@17.0.5/jdk.internal.misc.Signal$1.run(Signal.java:219)
        at java.base@17.0.5/java.lang.Thread.run(Thread.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
```

**GraalVM and environment:**
 - openjdk 17.0.5 2022-10-18
 - OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)
 - OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)
 - OS: CentOS7
 - Architecture: AMD64

I searched for related content , and the JVM seems to judge whether to register SIGHUP processing by whether the signal is ignored. What should I do?

**Comment by vicasong** [comment]
the demo tool repository: https://github.com/vicasong/demo-graalvm

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reporting this issue, we'll take a look into it and get back to you shortly.

**Comment by teshull** [comment]
This was fixed by #6447 and will be part of the 23.1 release
